mergify_engine/tests/unit/test_clients.py

```diff
@@ -28,44 +28,78 @@
+    httpserver: httpserver.HTTPServer,
+    endpoint: str,
+    response: dict[str, object],
+    http_client: github.AsyncGithubInstallationClient,
+) -> None:
+    httpserver.expect_request(endpoint).respond_with_json(response)
+
+    httpserver.expect_request(
+        "/app/installations/12345/access_tokens"
+    ).respond_with_json({"token": "<app_token>", "expires_at": "2100-12-31T23:59:59Z"})
+
+    httpserver.expect_request(
+        "/", headers={"Authorization": "token <installation-token>"}
+    ).respond_with_json({"work": True}, status=200)
+
+    ret = await http_client.get(httpserver.url_for("/"))
+
+    assert ret.json()["work"]
+    assert http_client.auth.owner == "testing"
+    assert http_client.auth.owner_id == 12345
+    assert len(httpserver.log) == 3
+
+    httpserver.check_assertions()
+
+
 @mock.patch.object(github.CachedToken, "STORAGE", {})
 @pytest.mark.asyncio
-async def test_client_installation_token(httpserver: httpserver.HTTPServer) -> None:
+async def test_client_installation_token_with_owner_name(
+    httpserver: httpserver.HTTPServer,
+) -> None:
     with mock.patch(
         "mergify_engine.config.GITHUB_API_URL",
         httpserver.url_for("/")[:-1],
     ):
-        httpserver.expect_request("/users/owner/installation").respond_with_json(
-            {
-                "id": 12345,
-                "target_type": "User",
-                "permissions": {
-                    "checks": "write",
-                    "contents": "write",
-                    "pull_requests": "write",
-                },
-                "account": {"login": "testing", "id": 12345},
-            }
-        )
-        httpserver.expect_request(
-            "/app/installations/12345/access_tokens"
-        ).respond_with_json(
-            {"token": "<app_token>", "expires_at": "2100-12-31T23:59:59Z"}
-        )
-
-        httpserver.expect_request(
-            "/", headers={"Authorization": "token <installation-token>"}
-        ).respond_with_json({"work": True}, status=200)
 
         async with github.AsyncGithubInstallationClient(
             github.get_auth(github_types.GitHubLogin("owner"))
         ) as client:
-            ret = await client.get(httpserver.url_for("/"))
-            assert ret.json()["work"]
+            _httpserver_connect(
+                httpserver, "/users/owner/installation", gh_fake_response, client
+            )
 
-    assert len(httpserver.log) == 3
 
-    httpserver.check_assertions()
+@mock.patch.object(github.CachedToken, "STORAGE", {})
+@pytest.mark.asyncio
+async def test_client_installation_token_with_owner_id(
+    httpserver: httpserver.HTTPServer,
+) -> None:
+    with mock.patch(
+        "mergify_engine.config.GITHUB_API_URL",
+        httpserver.url_for("/")[:-1],
+    ):
+
+        async with github.AsyncGithubInstallationClient(
+            github.get_auth(owner_id=github_types.GitHubAccountIdType(12345))
+        ) as client:
+            _httpserver_connect(
```

Comment:
`async def` function have to be awaited to be executed. (eg: `await _httpserver_connect(...)`)